### PR TITLE
[Merged by Bors] - feat: binomial theorem for subtraction

### PR DESCRIPTION
**Pull request by ahhwuhu**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/LegendrePoly.lean
Comment (YaelDillies) [inline Mathlib/Analysis/LegendrePoly.lean:33]:
```suggestion
  C n!⁻¹ * derivative^[n] (X ^ n * (1 - X) ^ n)
```

File: Mathlib/Analysis/LegendrePoly.lean
Comment (YaelDillies) [inline Mathlib/Analysis/LegendrePoly.lean:117]:
```suggestion
lemma legendre_eval_symm (n : ℕ) (x : ℝ) :
    eval x (Legendre n) = (-1) ^ n * eval (1 - x) (Legendre n) := by
```

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (YaelDillies) [inline Mathlib/Data/Nat/Choose/Sum.lean:80]:
```suggestion
    (x - y) ^ n = ∑ m ∈ range (n + 1), n.choose m • x ^ m * (- 1) ^ (n - m) * y ^ (n - m) := by
```

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (YaelDillies) [inline Mathlib/Data/Nat/Choose/Sum.lean:81]: Please don't use lemmas from tactic implementations
```suggestion
  rw [sub_eq_add_neg, add_pow]
```

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (YaelDillies) [inline Mathlib/Data/Nat/Choose/Sum.lean:1]: Can you make this be a separate PR?

Comment (ahhwuhu) [reply Mathlib/Data/Nat/Choose/Sum.lean:1]: okk

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (YaelDillies) [inline Mathlib/Data/Nat/Choose/Sum.lean:80]:
```suggestion
    (x - y) ^ n = ∑ m ∈ range (n + 1), (-1) ^ (m + n) * n.choose m • x ^ m * y ^ (n - m) := by
```
might be more canonical?

Comment (ahhwuhu) [reply Mathlib/Data/Nat/Choose/Sum.lean]: ok! I get it!

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (YaelDillies) [inline Mathlib/Data/Nat/Choose/Sum.lean:95]: I don't really see the value in this lemma. You get this just by applying `sub_pow` and rewriting the terms in the obvious fashion. Can you inline it in your application?

Comment (ahhwuhu) [reply Mathlib/Data/Nat/Choose/Sum.lean:95]: emmmmm. I get it! It must be ok!

Comment (ahhwuhu) [reply Mathlib/Data/Nat/Choose/Sum.lean:95]: I worried that my proof is too long, so I separate it. ok. I'll inline it in my lemma.

**Comment by ahhwuhu** [comment]
ok! I fix all problem! What should I do next?

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (YaelDillies) [inline Mathlib/Data/Nat/Choose/Sum.lean:91]:
```suggestion
  congr! 1 with m hm
  have : (-1 : R) ^ (n - m) = (-1) ^ (n + m) := by
    rw [mem_range] at hm
    simp [show n + m = n - m + 2 * m by omega, pow_add]
  rw [neg_pow, this]
  ring
```

Comment (ahhwuhu) [reply Mathlib/Data/Nat/Choose/Sum.lean]: thanks!

**Comment by ahhwuhu** [comment]
Thank you for your suggestion. I learn a lot. Then how can I merge this PR?

**Review by YaelDillies — APPROVED** [review]
Neither of us can merge this PR. We need the approval of a maintainer. Luckily, I have mystical powers that let me invoke one solely by ushering the words

maintainer merge

**Comment by ahhwuhu** [comment]
> Neither of us can merge this PR. We need the approval of a maintainer. Luckily, I have mystical powers that let me invoke one solely by ushering the words
> 
> maintainer merge

hahahahahaha! That's so cool! I'm so lucky to meet you to review my PR and I learn so much from you! Thank you very much Combinatorics math genius!!!

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (b-mehta) [inline Mathlib/Data/Nat/Choose/Sum.lean:79]:
```suggestion
theorem sub_pow [CommRing R] (x y : R) (n : ℕ) :
```

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (b-mehta) [inline Mathlib/Data/Nat/Choose/Sum.lean:80]:
```suggestion
    (x - y) ^ n = ∑ m ∈ range (n + 1), (-1) ^ (m + n) * x ^ m * y ^ (n - m) * n.choose m  := by
```
for consistency with the previous version

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (b-mehta) [inline Mathlib/Data/Nat/Choose/Sum.lean:13]: Do you still need this import?

**Review by b-mehta — APPROVED** [review]

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (ahhwuhu) [reply Mathlib/Data/Nat/Choose/Sum.lean:13]: oh! I don't need.

**Comment by b-mehta** [comment]
bors merge

**Pull request closed without merging**
